src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy };
+        Object.assign(stateCopy, obj.extraData);
+        break;
+      case 'removeProperties':
+        stateCopy = { ...stateCopy };
```

Comment:
if making a copy on `.push` - no need to copy on each `case`

Reply:
✅